### ISSUE TITLE
Add colorful message to_s() to threads to make it humanly pleasing and appealing

### ISSUE THREAD

**Issue by palladius** [patterns-ai-core/langchainrb#756]
Maybe this can help your demo Andrei. I've done this for my demo in Verona and 4mo after its still good code :)

```ruby
require 'colorize'

module Langchain
  module Messages
    class GoogleGeminiMessage
      def to_s
        # 💬
        # fun call
        if tool_calls.any?
          # for every tool_calls print one line:
          # 🤖 [model] 🛠️ [1/2] 🛠️  {"name"=>"news_retriever__get_top_headlines", ..
          # 🤖 [model] 🛠️ [2/2] 🛠️  {"name"=>"....",
          #return # test, should just be "\n"
          tool_calls.enum_for(:each_with_index).map do |tool_call, ix|
            return "🤖 [#{role}] 🛠️ [#{ix + 1}/#{tool_calls.count}] 🛠️  #{begin
              tool_call['functionCall'].to_s.force_encoding('UTF-8').colorize(:gray)
            rescue StandardError
              $ERROR_INFO
            end}"
          end.join("\n")
          # return "🤖 [#{role}] 🛠️ #{self.tool_calls[0]['functionCall'].to_s.force_encoding("UTF-8").colorize(:gray) rescue $!}" if self.tool_calls.any?
        end

        redaction_length = 3000
        sanitized_content = content.force_encoding('UTF-8').strip rescue content 
        if sanitized_content.length > redaction_length
          sanitized_content = "#{sanitized_content.first(redaction_length)}.. (🤥 redacted)"
        end

        # model
        return "🤖 [modl] 💬 #{sanitized_content.colorize :cyan}" if role == 'model'
        # user
        return "🧑 [user] 💬 #{sanitized_content.colorize :yellow}" if role == 'user'
        # function 🖩 would be great but too small 🧮 this is nice
        if role == 'function'
          return "🔢 [func] 🛠️  #{tool_call_id.force_encoding('UTF-8').colorize :white} => #{sanitized_content.colorize :gray}"
        end
        puts("Something weird happened here! role=#{role} #{self.inspect}")
        # if everything else fails...
        inspect # :status, :code, :message, ...
      end
    end
  end
end

```

Demo:

```
messages = assistant.add_message_and_run content: "Whats the weather like in San francisco, CA?",  auto_tool_execution: true
messages.each do |m|
  puts(" - #{m}")
end

==>

 - 🧑 [user] 💬 Whats the weather like in San francisco, CA?
 - 🤖 [model] 🛠️ [1/1] 🛠️  {"name"=>"langchain_tool_weather__get_current_weather", "args"=>{"state_code"=>"CA", "city"=>"San Francisco"}}
 - 🔢 [func] 🛠️  langchain_tool_weather__get_current_weather => Location not found
 - 🤖 [modl] 💬 I'm sorry, I don't have any information about the weather in San Francisco, CA.
```

**Comment by palladius** [comment]
![image](https://github.com/user-attachments/assets/7dc211d2-baa6-40a4-a9c6-e06b2ee06b40)

Shows the colorfulness.

**Comment by palladius** [comment]
Andrei if you like the idea I can change gem (from colorize to rainbow) and send a fix. You tell me

**Comment by andreibondarev** [comment]
> Andrei if you like the idea I can change gem (from colorize to rainbow) and send a fix. You tell me

We're already using the rainbow gem!

**Comment by palladius** [comment]
Let me rephrase it. Would you like it if I add a generic `pretty_print()` for a GoogleMessage or better any LLM BaseMessage in a PR?
If you find it silly or useless i wont spend that 30m doing it.
